src/town_cmd.cpp

```diff
@@ -2361,15 +2352,10 @@ static inline bool TownLayoutAllows2x2HouseHere(Town *t, TileIndex tile)
 	TileIndexDiffC grid_pos = TileIndexToTileIndexDiffC(t->xy, tile);
 
 	switch (t->layout) {
-		case TL_2X2_GRID:
-			grid_pos.x %= 3;
-			grid_pos.y %= 3;
-			if ((grid_pos.x != 2 && grid_pos.x != -1) ||
-				(grid_pos.y != 2 && grid_pos.y != -1)) return false;
-			break;
-
-		case TL_3X3_GRID:
-			if ((grid_pos.x & 3) < 2 || (grid_pos.y & 3) < 2) return false;
+		case TL_GRID:
+			grid_pos.x = Mod(grid_pos.x, t->spacing + 1);
```

Comment:
Not a fan of this weird mod thing. Why is it that `%` doesn't work / produce desirable results ?

Reply:
`%` rounds towards 0, whereas this `Mod` function rounds towards -infinity, which does different things for negative numbers.

using plain `%` would thus create special cases

Reply:
Could a lambda be used to avoid old c-style library functions being added for this trivial esoteric case?